### PR TITLE
Refactor to use expectExpression in analyzer.

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: libsolidity/inlineasm/AsmAnalysis.cpp [hunk 1]
```diff
@@ -349,6 +335,25 @@ bool AsmAnalyzer::operator()(Block const& _block)
...
+{
+	bool success = true;
+	int const initialHeight = m_stackHeight;
+	if (!boost::apply_visitor(*this, _statement))
+		success = false;
+	if (m_stackHeight - initialHeight != 1)
+	{
+		m_errorReporter.typeError(
+			locationOf(_statement),
+			"Expected instruction(s) to deposit one item to the stack but did deposit " +
```
Comment (axic) [inline libsolidity/inlineasm/AsmAnalysis.cpp]: comma missing before `but` and `one` -> `1`

File: libsolidity/inlineasm/AsmAnalysis.cpp [hunk 2]
```diff
@@ -349,6 +335,25 @@ bool AsmAnalyzer::operator()(Block const& _block)
...
+			boost::lexical_cast<string>(m_stackHeight - initialHeight) +
+			" items."
```
Comment (axic) [inline libsolidity/inlineasm/AsmAnalysis.cpp]: `items` -> `item(s)`

(Assuming tests check for this message.)

File: libsolidity/inlineasm/AsmAnalysis.cpp (hunk 1 above)
Comment (axic) [reply libsolidity/inlineasm/AsmAnalysis.cpp]: Seems like the text is not checked against in the tests, so don't mind if it is `one` or `1`, but would be nice to change `instruction(s)` to `expression`.

Comment (chriseth) [reply libsolidity/inlineasm/AsmAnalysis.cpp]: In inline assembly, they don't have to be expressions.

Comment (chriseth) [reply libsolidity/inlineasm/AsmAnalysis.cpp]: ok, but yeah, probably better

Comment (chriseth) [inline libsolidity/inlineasm/AsmAnalysis.cpp]: ok, but probably better

File: libsolidity/inlineasm/AsmAnalysis.cpp (hunk 2 above)
Comment (chriseth) [reply libsolidity/inlineasm/AsmAnalysis.cpp]: haha, I thought about this :-)
The check is always against one and this about a failure, so it is either `0 items` or `2 items` or more :-)

File: libsolidity/inlineasm/AsmAnalysis.cpp [hunk 3]
```diff
@@ -349,6 +335,25 @@ bool AsmAnalyzer::operator()(Block const& _block)
...
+{
+	bool success = true;
+	int const initialHeight = m_stackHeight;
+	if (!boost::apply_visitor(*this, _statement))
+		success = false;
+	if (m_stackHeight - initialHeight != 1)
+	{
+		m_errorReporter.typeError(
+			locationOf(_statement),
+			"Expected expression to return one item to the stack but did return " +
```
Comment (axic) [inline libsolidity/inlineasm/AsmAnalysis.cpp]: Can you please add back the comma before `but`? :)

**Review by axic — APPROVED** [review]
Looks good apart that tiny text change.

File: libsolidity/inlineasm/AsmAnalysis.cpp (hunk 3 above)
Comment (chriseth) [reply libsolidity/inlineasm/AsmAnalysis.cpp]: I swear I typed it!

File: libsolidity/inlineasm/AsmAnalysis.cpp (hunk 2 above)
Comment (axic) [reply libsolidity/inlineasm/AsmAnalysis.cpp]: True true, since it is limited to 1 now :)

**Pull request merged**